server/routerlicious/packages/services/src/mongodb.ts

```diff
@@ -455,21 +457,36 @@ export class MongoCollection<T> implements core.ICollection<T>, core.IRetryable
 			}
 		}
 
-		if (totalCount < this.apiMinimumCountToEnableTermination) {
-			// too little requests to consider
-			return;
-		}
 		const failureRate = totalFailedCount / totalCount;
+		const logProperties = {
+			failureRate,
+			totalCount,
+			totalFailedCount,
+			apiFailureRateTerminationThreshold: this.apiFailureRateTerminationThreshold,
+			apiMinimumCountToEnableTermination: this.apiMinimumCountToEnableTermination,
+			consecutiveFailedCount: this.consecutiveFailedCount,
+			consecutiveFailedThresholdForLowerTotalRequests:
+				this.consecutiveFailedThresholdForLowerTotalRequests,
+		};
 		if (failureRate <= this.apiFailureRateTerminationThreshold) {
-			// OK, no need to terminate
+			this.consecutiveFailedCount = 0;
 			return;
 		}
 
+		this.consecutiveFailedCount++;
+		if (
+			totalCount < this.apiMinimumCountToEnableTermination &&
+			this.consecutiveFailedCount < this.consecutiveFailedThresholdForLowerTotalRequests
+		) {
+			Lumberjack.warning("Total could didn't met min threshold", logProperties);
```

Comment:
typo - "Total **count** didn't **meet** min threshold". Also would be nice to add some context eg: total count for mongo failures.

Reply:
Corrected the logging message. total count for mongo failures is being reported in property field already